tests/lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,221 @@
+        },
+        {
+            code:
+                "Object.assign({ foo: 'bar' }, Object.assign({ bar: 'foo' }, Object.assign({}, { superNested: 'butwhy' })))",
+            output: "{foo: 'bar', bar: 'foo', superNested: 'butwhy'}",
+            parserOptions,
+            errors: [
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                },
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                },
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Mix spread in argument
+        {
+            code: "Object.assign({ foo: 'bar', ...bar }, baz)",
+            output: "{foo: 'bar', ...bar, ...baz}",
+            parserOptions,
+            errors: [
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Object shorthand
+        {
+            code: "Object.assign({}, { foo, bar, baz })",
+            output: "{foo, bar, baz}",
+            parserOptions,
+            errors: [
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Objects with computed properties
+        {
+            code: "Object.assign({}, { [bar]: 'foo' })",
+            output: "{[bar]: 'foo'}",
+            parserOptions,
+            errors: [
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Objects with spread properties
+        {
+            code: "Object.assign({ ...bar }, { ...baz })",
+            output: "{...bar, ...baz}",
+            parserOptions,
+            errors: [
+                {
+                    message: defaultErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        /*
+         * This is a special case where Object.assign is called with a single argument
+         * and that argument is an object expression. In this case we warn and display
+         * a message to use an object literal instead.
+         */
+        {
+            code: "Object.assign({})",
+            output: "{}",
+            errors: [
+                {
+                    message: useObjLiteralErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({ foo: bar })",
+            output: "{ foo: bar }",
+            errors: [
+                {
+                    message: useObjLiteralErrorMessage,
+                    type: "CallExpression"
+                }
+            ]
+        }
```

Comment:
Would you add tests for comments and parentheses? I expect comments to be kept.

For example:

```
Object.assign(
    {},
    // a comment
    foo,
    // another comment
    { a: true }
)
```

```
Object.assign(
    {},
    ({ a }),
    b ? c : { d },
    (e, f)
)
```